src/app/frontend/logs/logs_factory.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Factory function for logs panel management.
+ * @return {{registerListener: Function, notifyTextColorChange: Function}}
+ */
+export default function logsFactory() {
+  /**
+   * @type {(Object|null)} logsListener
+   */
+  let logsListener = null;
+
+  /**
+   * Register listener function.
+   * @param {!Object} listener
+   */
+  let registerListener = function(listener) { logsListener = listener; };
```

Comment:
Instead of notify/listener pattern you could have just a setter and getter here. This would simplify logic, as you only get the value in client and set in producer and make it possible to have multiple client, which cannot be done now. What do you think?